cmake/DaemonGame.cmake

```diff
@@ -75,41 +76,80 @@ function(GAMEMODULE)
+                        SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}
+                        BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/${vm}-${NACL_TARGET}
+                        CMAKE_GENERATOR ${VM_GENERATOR}
+                        CMAKE_ARGS
+                            -DFORK=2
+                            -DCMAKE_TOOLCHAIN_FILE=${Daemon_SOURCE_DIR}/cmake/toolchain-saigo.cmake
+                            -DDAEMON_DIR=${Daemon_SOURCE_DIR}
+                            -DDEPS_DIR=${DEPS_DIR}
+                            -DBUILD_GAME_NACL=ON
+                            -DUSE_NACL_SAIGO=ON
+                            -DNACL_TARGET=${NACL_TARGET}
+                            -DSAIGO_ARCH=${SAIGO_ARCH}
+                            -DBUILD_GAME_NATIVE_DLL=OFF
+                            -DBUILD_GAME_NATIVE_EXE=OFF
+                            -DBUILD_CLIENT=OFF
+                            -DBUILD_TTY_CLIENT=OFF
+                            -DBUILD_SERVER=OFF
+                            ${inherited_option_args}
+                        INSTALL_COMMAND ""
+                    )
+                endforeach()
+            else()
+                # Workaround a bug where CMake ExternalProject lists-as-args are cut on first “;”
+                string(REPLACE ";" "," NACL_TARGETS_STRING "${NACL_TARGETS}")
 
+                ExternalProject_Add(${vm}
+                    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}
+                    BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/${vm}
+                    CMAKE_GENERATOR ${VM_GENERATOR}
+                    CMAKE_ARGS
+                        -DFORK=2
+                        -DCMAKE_TOOLCHAIN_FILE=${Daemon_SOURCE_DIR}/cmake/toolchain-pnacl.cmake
+                        -DDAEMON_DIR=${Daemon_SOURCE_DIR}
+                        -DDEPS_DIR=${DEPS_DIR}
+                        -DBUILD_GAME_NACL=ON
+                        -DNACL_TARGETS_STRING=${NACL_TARGETS_STRING}
+                        -DBUILD_GAME_NATIVE_DLL=OFF
+                        -DBUILD_GAME_NATIVE_EXE=OFF
+                        -DBUILD_CLIENT=OFF
+                        -DBUILD_TTY_CLIENT=OFF
+                        -DBUILD_SERVER=OFF
+                        ${inherited_option_args}
+                    INSTALL_COMMAND ""
+                )
+                ExternalProject_Add_Step(${vm} forcebuild
+                    COMMAND ${CMAKE_COMMAND} -E remove
+                        ${CMAKE_CURRENT_BINARY_DIR}/${vm}-prefix/src/${vm}-stamp/${vm}-configure
+                    COMMENT "Forcing build step for '${vm}'"
+                    DEPENDEES build
+                    ALWAYS 1
+                )
+            endif()
         endif()
     else()
         if (FORK EQUAL 2)
-            # Put the .nexe and .pexe files in the same directory as the engine
-            set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/..)
+            if(USE_NACL_SAIGO)
+                set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
```

Comment:
Why is this different? We want them to be in the same directory as the engine so that VM type 1 can be used.

Reply:
If my memory is right (I wrote this 1 year ago), with PNaCl we can output `sgame.pexe` to the main build directory because it's the same for all platforms, then `pnacl_finalize` translate it to `sgame-amd64.nexe`, `sgame-i686.nexe`, etc.

But with Saigo we have one subproject per platform, and it compiles to nexe directly so every subproject output an `sgame.nexe` file with the exact same name and then every platform would overwrite the file if we output it to the main build directory. So we output it the `sgame.nexe` file to every subproject build directory and then `saigo_finalize` rename them to `sgame-amd64.nexe`, `sgame-i686.nexe`, etc. into the main build directory.

You can see below that `saigo_finalize` gets `${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/..` (which was to `${CMAKE_BINARY_DIR}/..`) while `pnacl_finalize` gets `${CMAKE_RUNTIME_OUTPUT_DIRECTORY}` (which was already set to `${CMAKE_BINARY_DIR}/..`).